temporalio/worker/workflow.py

```diff
@@ -214,10 +220,15 @@ async def _handle_activation(
 
         # If there is a remove-from-cache job, do so
         if remove_job:
-            logger.debug(
-                f"Evicting workflow with run ID {act.run_id}, message: {remove_job.remove_from_cache.message}"
-            )
-            del self._running_workflows[act.run_id]
+            if act.run_id in self._running_workflows:
+                logger.debug(
+                    f"Evicting workflow with run ID {act.run_id}, message: {remove_job.remove_from_cache.message}"
+                )
+                del self._running_workflows[act.run_id]
+            else:
+                logger.debug(
+                    f"Eviction request on unknown workflow with run ID {act.run_id}, message: {remove_job.remove_from_cache.message}"
+                )
```

Comment:
Any reason for eagerly evaluating the log strings?

Reply:
No, I need to fix this (though obviously the perf benefit is negligible)

Reply:
Done